src/core/common/message.hpp

```diff
@@ -90,6 +90,7 @@ struct MessageListEntry
 struct BufferHeader
 {
     class Buffer *mNext;  ///< A pointer to the next Message buffer.
+    otContext *mContext;  ///< A pointer to the OpenThread context.
```

Comment:
Should we save the context with every buffer or should we pass the context into every Message function?